src/nvim/tui/tui.c

```diff
@@ -2000,7 +2000,7 @@ static const char *tui_tk_ti_getstr(const char *name, const char *value,
   } else if (strequal(name, "key_dc")) {
     DLOG("libtermkey:kdch1=%s", value);
     // Vim: "If <BS> and <DEL> are now the same, redefine <DEL>."
-    if (value != NULL && strequal(stty_erase, value)) {
+    if (value != NULL && value != (char *)-1 && strequal(stty_erase, value)) {
```

Comment:
This looks weird. Is `-1` really passed as a pointer? Or should it be `*value != (char)-1` ?

Reply:
Hello!

Yes, the -1 is really passed as a pointer. It is documented in man 3 tigetstr.
Here is a link:
https://linux.die.net/man/3/tigetstr

the relevant part is:
The tigetstr routine returns the value (char *)-1 if capname is not a string capability, or 0 if it is canceled or absent from the terminal description. 

Also, it seems that in this situation NULL should be returned, but in practice -1 is returned somehow.

Most probably there is another problem in curses, but handling this, otherwise acceptable return value would make sense.

If not this way, then by signaling an error.